core/src/test/java/org/elasticsearch/rest/CorsOriginsIT.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.rest;
+
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.test.ESIntegTestCase;
+import org.elasticsearch.test.ESIntegTestCase.ClusterScope;
+import org.elasticsearch.node.Node;
+import org.elasticsearch.test.rest.client.http.HttpResponse;
+
+import static org.elasticsearch.http.netty.NettyHttpServerTransport.SETTING_CORS_ALLOW_CREDENTIALS;
+import static org.elasticsearch.http.netty.NettyHttpServerTransport.SETTING_CORS_ALLOW_METHODS;
+import static org.elasticsearch.http.netty.NettyHttpServerTransport.SETTING_CORS_ALLOW_ORIGIN;
+import static org.elasticsearch.http.netty.NettyHttpServerTransport.SETTING_CORS_ENABLED;
+
+/**
+ * Test CORS where the allow origin value are string literals.
+ */
+@ClusterScope(scope = ESIntegTestCase.Scope.SUITE, numDataNodes = 1)
+public class CorsOriginsIT extends CorsRegexIT {
+
+    @Override
+    protected Settings nodeSettings(int nodeOrdinal) {
+        final String corsValue = "http://localhost:9200";
+        return Settings.settingsBuilder()
+                   .put(super.nodeSettings(nodeOrdinal))
+                   .put(SETTING_CORS_ALLOW_ORIGIN, corsValue)
+                   .put(SETTING_CORS_ALLOW_METHODS, "get, options, post")
+                   .put(SETTING_CORS_ALLOW_CREDENTIALS, true)
+                   .put(SETTING_CORS_ENABLED, true)
+                   .put(Node.HTTP_ENABLED, true)
+                   .build();
+    }
+
+    @Override
+    public void testThatRegularExpressionWorksOnMatch() throws Exception {
+        String corsValue = "http://localhost:9200";
```

Comment:
same here, a unit test starting the netty transport should be sufficient and much faster?